### PR TITLE
Add service catalog testing to daily runs

### PR DESCRIPTION
**Pull request by jpeeler**
This is nearly identical to the existing "/test service-catalog" execution in test_pull_request_origin_service_catalog.yml.

### CODE REVIEW

File: sjb/config/test_cases/test_branch_origin_service_catalog.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+parent: 'common/test_cases/origin_built_release.yml'
+overrides:
+  junit_analysis: False
+  timer: 'H H * * *'
+  email:
+    - skuznets@redhat.com
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_service_catalog.yml]: no, please

**Comment by jpeeler** [comment]
This is ready for last review/merging now, @stevekuznetsov .

File: sjb/config/test_cases/test_branch_origin_service_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      script: |-
+        ./_output/local/bin/linux/amd64/oc cluster up --version=latest --service-catalog
+    - type: "script"
+      title: "Run Service Catalog E2E"
+      repository: "origin"
+      timeout: 1800
+      script: |-
+        sudo chmod a+rw /var/lib/origin/openshift.local.config/master/admin.kubeconfig
+        export KUBECONFIG=/var/lib/origin/openshift.local.config/master/admin.kubeconfig
+        ./_output/local/bin/linux/amd64/oc login -u system:admin
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_service_catalog.yml]: This `oc` should be in the `$PATH` if you care about making this a little shorter

File: sjb/config/test_cases/test_branch_origin_service_catalog.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      repository: "origin"
+      timeout: 1800
+      script: |-
+        hack/update-generated-bindata.sh
+    - type: "script"
+      title: "build Origin OC binary with latest RBAC"
+      repository: "origin"
+      timeout: 1800
+      script: |-
+        OS_BUILD_ENV_PULL_IMAGE=true OS_BUILD_ENV_PRESERVE=_output/local/bin/linux/amd64/oc hack/env make build WHAT=cmd/oc
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_service_catalog.yml]: The point of using the `build` stage as the base is to not require a re-build of any Origin bits -- that image is updated after every merge to `master`

File: sjb/config/test_cases/test_branch_origin_service_catalog.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
+---
+parent: 'common/test_cases/origin_built_release.yml'
+overrides:
+  junit_analysis: False
+  timer: 'H H * * *'
+  email:
+    - jpeeler@redhat.com
+extensions:
+  actions:
+    - type: "forward_parameters"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_service_catalog.yml]: i think you can ditch this one?

File: sjb/config/test_cases/test_branch_origin_service_catalog.yml (hunk 2 above)
Comment (jpeeler) [reply sjb/config/test_cases/test_branch_origin_service_catalog.yml]: It actually didn't end up working without the full path, so I left it.

**Comment by jpeeler** [comment]
@stevekuznetsov should be good to go now!

**Pull request merged**